File: backend/src/main/java/cl/supletanes/supletanes_app/producto/repository/ProductoRepository.java
package cl.supletanes.supletanes_app.producto.repository;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;

import cl.supletanes.supletanes_app.producto.model.Producto;

public interface ProductoRepository extends JpaRepository<Producto, Long>{
    List<Producto> findByCategoria(String categoria);
}

File: frontend/app/src/main/java/com/example/supletanes/repository/ProductoRepository.kt
package com.example.supletanes.repository

import com.example.supletanes.data.api.ProductoApiService
import com.example.supletanes.data.model.Producto
import com.example.supletanes.data.model.ProductoRequest

class ProductoRepository(private val apiService: ProductoApiService) {

    suspend fun crearProducto(request: ProductoRequest): Result<Producto> {
        return try {
            val response = apiService.crearProducto(request)
            if (response.isSuccessful && response.body() != null) {
                Result.success(response.body()!!)
            } else {
                Result.failure(Exception("Error al crear producto: código ${response.code()}"))
            }
        } catch (e: Exception) {
            Result.failure(e)
        }
    }
}